### PR TITLE
Sdk next major version

### PR DESCRIPTION
**Pull request by plmercereau**
Includes:
- [x] [remove deprecated methods](https://github.com/nhost/nhost/pull/1233)
- [x] [replace axios by cross-fetch](https://github.com/nhost/nhost/pull/1351)
- [x] [new GraphQL client](https://github.com/nhost/nhost/pull/1587)

Fixes #1464

### CODE REVIEW

File: .changeset/curvy-houses-remember.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+return <form onSubmit={handleSubmit}>
+            <input value={email} onChange={onChange={(event) => setEmail(event.target.value)}}  />
+       </form>
+```
+
+Instead, write:
+
+```tsx
+const [email] = useState('')
+const {changeEmail} = useChangeEmail(email)
```
Comment (elitan) [inline .changeset/curvy-houses-remember.md:37]: Should this be:


```js
const {changeEmail} = useChangeEmail()
```

without the `email` in the `useChangeEmail` hook?

**Review by elitan — APPROVED** [review]
One comment. Once fixed, looks good.

**Review by szilarddoro — APPROVED** [review]

File: .changeset/curvy-houses-remember.md (hunk 1 above)
Comment (szilarddoro) [reply .changeset/curvy-houses-remember.md:37]: I think it's fine. It's going to be easier to understand what the function does later in the code just by taking a look at it.

File: .changeset/curvy-houses-remember.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+return <form onSubmit={handleSubmit}>
+            <input value={email} onChange={onChange={(event) => setEmail(event.target.value)}}  />
+       </form>
+```
+
+Instead, write:
+
+```tsx
+const [email] = useState('')
```
Comment (szilarddoro) [inline .changeset/curvy-houses-remember.md:36]:
```suggestion
const [email, setEmail] = useState('')
```

File: .changeset/curvy-houses-remember.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+'@nhost/react': major
+---
+
+Remove deprecated signatures in React hooks
```
Comment (szilarddoro) [inline .changeset/curvy-houses-remember.md:5]: Our changelogs should not be this long. Turn this into a PR description instead and use a short description.

**Review by szilarddoro — CHANGES_REQUESTED** [review]

**Review by szilarddoro — APPROVED** [review]

**Pull request merged**